src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Collections.Generic;
+
+namespace Iot.Device.Tlc1543
+{
+    /// <summary>
+    /// Add documentation here
+    /// </summary>
+    public class Tlc1543 : IDisposable
+    {
+        #region Members
+
+        /// <summary>
+        /// Channel used between readings to charge up capacitors in ADC
+        /// </summary>
+        public Channel ChargeChannel = Channel.SelfTest512;
+        private readonly int _chipSelect;
+        private readonly int _dataOut;
+        private readonly int _address;
+        private readonly int _inputOutputClock;
+        private readonly int _endOfConversion;
+        private readonly int _channels = 11;
+        private GpioController _digital = new GpioController(PinNumberingScheme.Logical);
+        private bool _disposedValue;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543

```

Comment:
Looking at the full code, it seems that this sensor can work as well with SPI. So would be nice as well to have a constructor and also the capacity to have it working with SPI if that's the case.

Reply:
I'm not experienced in SPI standard, so wasn't looking for that, but also I haven't seen any mention of proper SPI support in documentation. This may be something to consider in the future.

Reply:
Fair enough. Let's keep then this binding with the GPIO only support so far.